src/lock.rs

```diff
@@ -108,32 +123,38 @@ impl Lock {
         name: &str,
         version: &str,
         bump_func: Box<BumpFunc>,
-    ) -> Result<String, Error> {
+    ) -> Result<(bool, String), Error> {
         let mut res = String::new();
-        let mut num_changes = 0;
+        let mut num_matches = 0;
+        let mut changed = false;
         for (i, line) in self.contents.lines().enumerate() {
             let bumped_line = (bump_func)(line, name, version);
             let bumped_line = bumped_line.map_err(|e| Error::MalformedLock {
                 line: i + 1,
                 details: e.details,
             })?;
-            if bumped_line != line {
-                num_changes += 1;
-            }
-            res.push_str(&bumped_line);
+            if let Some(bumped_line) = bumped_line {
+                num_matches += 1;
+                res.push_str(&bumped_line);
+                if bumped_line != line {
+                    changed = true;
+                }
+            } else {
+                res.push_str(line);
+            };
             res.push_str("\n");
```

Comment:
Gratuitous suggestion: Instead of doing push_str twice it's actually a *tiny* bit shorter/more efficient to just use format!:
```
    res = format!("{}\n", bumped_line);
} else {
   res = format!("{}\n", line);
```

(imo format is slightly cooler in general because if this weren't in a for loop you could make res non-mutable!)